### DISCUSSION TITLE
Levensfasen - uitwerking 5 - Geplande objecten in aparte registratie

### DISCUSSION THREAD

**Discussion by pmaria** [Geonovum/disgeo-imsor#34] [General]
Deze modelleringen zijn uitwerkingen  o.b.v. [deze casus](https://github.com/Geonovum/disgeo-imsor/tree/master/issues/15-levensfasen/uitwerkingen) waarbij er voor geplande objecten een aparte registratie wordt gehanteerd.

**Comment by pmaria** [comment]
In deze uitwerking wordt er onderscheid gemaakt tussen gegevens over geplande en gerealiseerde object door deze in aparte registraties op te nemen. Het enige verschil tussen de informatiemodellen in de registraties is de inhoud van de `Status` enumeratie.

![5-planobject-in-aparte-registratie](https://github.com/Geonovum/disgeo-imsor/raw/master/issues/15-levensfasen/uitwerkingen/5-planobject-in-aparte-registratie/5-planobject-in-aparte-registratie.png)

### Voordelen (wordt bijgewerkt adhv discussie)
- Informatiemodel is simpel (geen extra objecttypes nodig)
- Kan complexe situatie aan: meerdere plannen per object
- ?

### Nadelen (wordt bijgewerkt adhv discussie)
- Twee registraties nodig
- Integrale bevraging plan en werkelijkheid wordt complexer
- Onderscheid tussen of iets een plan is of niet is onduidelijk bij aanlevering
- ?

**Comment by lvdbrink** [comment]
Hmm, als je twee registraties nodig hebt, heb je dan ook twee informatiemodellen nodig? Ik zie wel een verschil, in het statussenlijstje... Dit lijkt me wel een nadeel.

**Comment by RiX012** [comment]
In essentie is dit toch gewoon alternatief 2a maar dan gespreid over 2 registraties? Op zich is dan ook het voordeel van 2a te noemen: "Objecttypes die niet te maken hebben met plannen hebben er geen last van".
Ik zie de voordelen van de registraties splitsen meer op een governance niveau dan op een informatiemodel niveau. Maar ik zou dat niet in deze discussie willen mengen (en daarmee 5 laten afvallen).

**Reply by pmaria** [reply]
Eens!

**Comment by RuudKathmann** [comment]
Deze oplossing lijkt mij ernstig strijdig met het uitgangspunt dat de SOR (die ene registratie dus) geschikt moet zijn voor tijdreizen.
Bij tijdreizen zou ik dan steeds tussen beide registratie moeten switchen.

**Reply by pmaria** [reply]
Tijdreizen kan mijns inziens nog steeds met twee registraties. Wanneer men alleen geïnteresseerd is in gerealiseerde objecten, of alleen geïnteresseerd is in geplande objecten is het waarschijnlijk zelfs vrij simpel. Maar, ook gecombineerd tijdreizen zou met een dienst bovenop deze registraties waarschijnlijk mogelijk zijn.

**Comment by lvdbrink** [comment]
Aanvullingen uit overleg 15 april: 

- De statuslijst is verschillend voor de 2 registraties, je kunt dus niet één precies hetzelfde informatiemodel gebruiken voor beide registraties, dit is een nadeel.
- Tijdreizen is moelijker, dit moet bv in bovenliggende service worden gerealiseerd. Dit valt volgens Lennart overigens wel mee, je maakt eigenlijk 2 subdomeinen.
- Je kunt meerdere plannen na elkaar hebben, via de tijdgeldigheid.
